sql/expression/distance.go

```diff
@@ -0,0 +1,110 @@
+package expression
+
+import (
+	"fmt"
+
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/types"
+)
+
+type DistanceType interface {
+	String() string
+	Eval(left []float64, right []float64) (float64, error)
+	CanEval(distanceType DistanceType) bool
+}
+
+type DistanceL2Squared struct{}
+
+var _ fmt.Stringer = DistanceL2Squared{}
+var _ DistanceType = DistanceL2Squared{}
+
+func (d DistanceL2Squared) String() string {
+	return "VEC_DISTANCE_L2_SQUARED"
+}
+
+func (d DistanceL2Squared) Eval(left []float64, right []float64) (float64, error) {
```

Comment:
Maybe add some unit tests for sanity `distance_test.go`

Reply:
Added unit test for sanity.